### PR TITLE
fixes for crystal 0.29.0

### LINKED ISSUE #337: Crystal 0.29.0 support
Crystal 0.29.0 is out and with breaking changes (https://github.com/crystal-lang/crystal/releases/tag/0.29.0).

@Blacksmoke16 and I discussed over Gitter DMs and we're aware of the following things that needs a fix:

- [ ] Json serialization - [broken by these changes](https://github.com/crystal-lang/crystal/pull/7818#issuecomment-498086324)

### PR DESCRIPTION
**Pull request by drujensen**
This fixes #337 the JSON serializers to support Crystal 0.29.0.  Thanks @bcardiff for your help.

### CODE REVIEW

**Comment by Blacksmoke16** [comment]
Well shit, that's all you had to do is change the type restriction to `: DB::Any`?

**Comment by drujensen** [comment]
@Blacksmoke16 and move them into `macro inherited`.  @bcardiff was close.

**Comment by Blacksmoke16** [comment]
Lol, nice one.

File: src/granite/base.cr [hunk 1]
```diff
@@ -43,16 +43,16 @@ abstract class Granite::Base
...
-    set_attributes(args.to_h)
-  end
+    def initialize(**args : DB::Any)
+      set_attributes(args.to_h)
+    end
 
-  def initialize(args : Hash(Symbol | String, DB::Any))
-    set_attributes(args)
-  end
+    def initialize(args : Hash(Symbol | String, DB::Any))
```
Comment (Blacksmoke16) [inline src/granite/base.cr]:
```suggestion
    def initialize(args : Hash(Symbol | String, Granite::Fields::Type))
```

File: src/granite/base.cr [hunk 2]
```diff
@@ -43,16 +43,16 @@ abstract class Granite::Base
       __process_transactions
       __process_migrator
     end
-  end
 
-  def initialize(**args : Object)
-    set_attributes(args.to_h)
-  end
+    def initialize(**args : DB::Any)
```
Comment (Blacksmoke16) [inline src/granite/base.cr]:
```suggestion
    def initialize(**args : Granite::Fields::Type)
```

**Comment by Blacksmoke16** [comment]
@drujensen I think you missed one in `base.cr`.  Also sorry, I just tried this.  Will have to add

```cr
{% elsif type.resolve <= Array %}
  @{{_name.id}} = value.as({{type.id}})
```

to `cast_to_field` to property cast the array to its expected type, which never had to happen before since it was handled by the DB driver I guess.

**Comment by drujensen** [comment]
@Blacksmoke16 fixed

**Pull request merged**